Binary compatibility broken by #648

### Issue description

#648 broke binary compatibility. 

```
./gradlew japiCmp
Type-safe project accessors is an incubating feature.

> Task :micronaut-micrometer-registry-cloudwatch:compileJava
Note: Creating bean classes for 1 type elements

> Task :micronaut-micrometer-registry-cloudwatch:japiCmp FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':micronaut-micrometer-registry-cloudwatch:japiCmp'.
> A failure occurred while executing me.champeau.gradle.japicmp.JApiCmpWorkAction
   > Detected binary changes.
         - current: micronaut-micrometer-registry-cloudwatch-5.3.0-SNAPSHOT.jar
         - baseline: micronaut-micrometer-registry-cloudwatch-5.2.0.jar.
     
     See failure report at file:///Users/sdelamo/github/micronaut-projects/micronaut-micrometer/micrometer-registry-cloudwatch/build/reports/binary-compatibility-micronaut-micrometer-registry-cloudwatch.html

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.4/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 12s
128 actionable tasks: 97 executed, 31 up-to-date

Publishing build scan...
https://ge.micronaut.io/s/d3nh35kyhfam4
```

--- a/config/accepted-api-changes.json
+++ b/config/accepted-api-changes.json
@@ -78,5 +78,185 @@
     "type": "io.micronaut.micrometer.annotation.processing.TimedSetAnnotationMapper",
     "member": "Implemented interface io.micronaut.inject.annotation.AnnotationMapper",
     "reason": "Provide a human readable reason for the change"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClient1$Definition$Reference",
+    "member": "Field $ANNOTATION_METADATA",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClient1$Definition$Reference",
+    "member": "Constructor io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClient1$Definition$Reference()",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClient1$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.annotation.AnnotationSource",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClient1$Definition$Reference",
+    "member": "Implemented interface io.micronaut.inject.BeanContextConditional",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClient1$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.annotation.AnnotationMetadataProvider",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClient1$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.beans.BeanInfo",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClient1$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.annotation.AnnotationMetadataDelegate",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClient1$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.annotation.AnnotationMetadata",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClient1$Definition$Reference",
+    "member": "Implemented interface io.micronaut.inject.QualifiedBeanType",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClient1$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.type.ArgumentCoercible",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClient1$Definition$Reference",
+    "member": "Implemented interface io.micronaut.inject.BeanType",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClient1$Definition$Reference",
+    "member": "Implemented interface io.micronaut.inject.BeanDefinitionReference",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClientBuilder0$Definition$Reference",
+    "member": "Field $ANNOTATION_METADATA",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClientBuilder0$Definition$Reference",
+    "member": "Constructor io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClientBuilder0$Definition$Reference()",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClientBuilder0$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.annotation.AnnotationSource",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClientBuilder0$Definition$Reference",
+    "member": "Implemented interface io.micronaut.inject.BeanContextConditional",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClientBuilder0$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.annotation.AnnotationMetadataProvider",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClientBuilder0$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.beans.BeanInfo",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClientBuilder0$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.annotation.AnnotationMetadataDelegate",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClientBuilder0$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.annotation.AnnotationMetadata",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClientBuilder0$Definition$Reference",
+    "member": "Implemented interface io.micronaut.inject.QualifiedBeanType",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClientBuilder0$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.type.ArgumentCoercible",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClientBuilder0$Definition$Reference",
+    "member": "Implemented interface io.micronaut.inject.BeanType",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchAsyncClientBuilder0$Definition$Reference",
+    "member": "Implemented interface io.micronaut.inject.BeanDefinitionReference",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchMeterRegistry2$Definition$Reference",
+    "member": "Field $ANNOTATION_METADATA",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchMeterRegistry2$Definition$Reference",
+    "member": "Constructor io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchMeterRegistry2$Definition$Reference()",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchMeterRegistry2$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.annotation.AnnotationSource",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchMeterRegistry2$Definition$Reference",
+    "member": "Implemented interface io.micronaut.inject.BeanContextConditional",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchMeterRegistry2$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.annotation.AnnotationMetadataProvider",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchMeterRegistry2$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.beans.BeanInfo",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchMeterRegistry2$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.annotation.AnnotationMetadataDelegate",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchMeterRegistry2$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.annotation.AnnotationMetadata",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchMeterRegistry2$Definition$Reference",
+    "member": "Implemented interface io.micronaut.inject.QualifiedBeanType",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchMeterRegistry2$Definition$Reference",
+    "member": "Implemented interface io.micronaut.core.type.ArgumentCoercible",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchMeterRegistry2$Definition$Reference",
+    "member": "Implemented interface io.micronaut.inject.BeanType",
+    "reason": "use AWS cloud watch client instead of default one"
+  },
+  {
+    "type": "io.micronaut.configuration.metrics.micrometer.cloudwatch.$CloudWatchMeterRegistryFactory$CloudWatchMeterRegistry2$Definition$Reference",
+    "member": "Implemented interface io.micronaut.inject.BeanDefinitionReference",
+    "reason": "use AWS cloud watch client instead of default one"
   }
 ]

--- a/micrometer-registry-cloudwatch/src/main/java/io/micronaut/configuration/metrics/micrometer/cloudwatch/CloudWatchMeterRegistryFactory.java
+++ b/micrometer-registry-cloudwatch/src/main/java/io/micronaut/configuration/metrics/micrometer/cloudwatch/CloudWatchMeterRegistryFactory.java
@@ -23,7 +23,6 @@ import io.micronaut.configuration.metrics.micrometer.ExportConfigurationProperti
 import io.micronaut.context.annotation.Bean;
 import io.micronaut.context.annotation.Factory;
 import io.micronaut.context.annotation.Requires;
-import io.micronaut.core.annotation.Internal;
 import io.micronaut.core.annotation.Nullable;
 import jakarta.inject.Inject;
 import jakarta.inject.Named;

--- a/micrometer-registry-cloudwatch/src/main/java/io/micronaut/configuration/metrics/micrometer/cloudwatch/CloudWatchMeterRegistryFactory.java
+++ b/micrometer-registry-cloudwatch/src/main/java/io/micronaut/configuration/metrics/micrometer/cloudwatch/CloudWatchMeterRegistryFactory.java
@@ -62,7 +62,7 @@ public class CloudWatchMeterRegistryFactory extends AwsClientFactory<CloudWatchC
     /**
      * @deprecated Use {@link CloudWatchMeterRegistryFactory(AwsCredentialsProviderChain, AwsRegionProviderChain, UserAgentProvider, AWSServiceConfiguration)} instead.
      */
-    @Deprecated
+    @Deprecated(since = "5.2.0", forRemoval = true)
     public CloudWatchMeterRegistryFactory() {
         this(AwsCredentialsProviderChain.builder().build(), new AwsRegionProviderChain(), null, null);
     }
@@ -71,7 +71,7 @@ public class CloudWatchMeterRegistryFactory extends AwsClientFactory<CloudWatchC
      *
      * @deprecated Use {@link #createAsyncBuilder()} instead.
      */
-    @Deprecated
+    @Deprecated(since = "5.2.0", forRemoval = true)
     CloudWatchAsyncClientBuilder cloudWatchAsyncClientBuilder() {
         return createAsyncBuilder();
     }
@@ -120,7 +120,7 @@ public class CloudWatchMeterRegistryFactory extends AwsClientFactory<CloudWatchC
      * @return A {@link CloudWatchAsyncClient}
      * @deprecated Use {{@link #asyncClient(CloudWatchAsyncClientBuilder)}} instead.
      */
-    @Deprecated
+    @Deprecated(since = "5.2.0", forRemoval = true)
     CloudWatchAsyncClient cloudWatchAsyncClient(CloudWatchAsyncClientBuilder builder) {
        return asyncClient(builder);
     }

--- a/micrometer-registry-cloudwatch/src/main/java/io/micronaut/configuration/metrics/micrometer/cloudwatch/CloudWatchMeterRegistryFactory.java
+++ b/micrometer-registry-cloudwatch/src/main/java/io/micronaut/configuration/metrics/micrometer/cloudwatch/CloudWatchMeterRegistryFactory.java
@@ -23,7 +23,9 @@ import io.micronaut.configuration.metrics.micrometer.ExportConfigurationProperti
 import io.micronaut.context.annotation.Bean;
 import io.micronaut.context.annotation.Factory;
 import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.Internal;
 import io.micronaut.core.annotation.Nullable;
+import jakarta.inject.Inject;
 import jakarta.inject.Named;
 import jakarta.inject.Singleton;
 import software.amazon.awssdk.auth.credentials.AwsCredentialsProviderChain;
@@ -50,11 +52,29 @@ public class CloudWatchMeterRegistryFactory extends AwsClientFactory<CloudWatchC
     public static final String CLOUDWATCH_ENABLED = CLOUDWATCH_CONFIG + ".enabled";
     public static final String CLOUDWATCH_DEFAULT_NAMESPACE = "micronaut";
 
+    @Inject
     protected CloudWatchMeterRegistryFactory(AwsCredentialsProviderChain credentialsProvider,
                                           AwsRegionProviderChain regionProvider,
                                           @Nullable UserAgentProvider userAgentProvider,
                                           @Nullable @Named(CloudWatchClient.SERVICE_NAME) AWSServiceConfiguration awsServiceConfiguration) {
         super(credentialsProvider, regionProvider, userAgentProvider, awsServiceConfiguration);
+    }
+
+    /**
+     * @deprecated Use {@link CloudWatchMeterRegistryFactory(AwsCredentialsProviderChain, AwsRegionProviderChain, UserAgentProvider, AWSServiceConfiguration)} instead.
+     */
+    @Deprecated
+    public CloudWatchMeterRegistryFactory() {
+        this(AwsCredentialsProviderChain.builder().build(), new AwsRegionProviderChain(), null, null);
+    }
+
+    /**
+     *
+     * @deprecated Use {@link #createAsyncBuilder()} instead.
+     */
+    @Deprecated
+    CloudWatchAsyncClientBuilder cloudWatchAsyncClientBuilder() {
+        return createAsyncBuilder();
     }
 
     @Override
@@ -96,14 +116,26 @@ public class CloudWatchMeterRegistryFactory extends AwsClientFactory<CloudWatchC
     }
 
     /**
-     * Create a CloudWatchMeterRegistry bean if global metrics are enabled
-     * and CloudWatch is enabled. Will be true by default when this
-     * configuration is included in project.
      *
-     * @param exportConfigurationProperties The export configuration
-     * @param cloudWatchAsyncClient  The CloudWatch async client
-     * @return A CloudWatchMeterRegistry
+     * @param builder A {@link CloudWatchAsyncClient} builder.
+     * @return A {@link CloudWatchAsyncClient}
+     * @deprecated Use {{@link #asyncClient(CloudWatchAsyncClientBuilder)}} instead.
      */
+    @Deprecated
+    CloudWatchAsyncClient cloudWatchAsyncClient(CloudWatchAsyncClientBuilder builder) {
+       return asyncClient(builder);
+    }
+
+
+        /**
+         * Create a CloudWatchMeterRegistry bean if global metrics are enabled
+         * and CloudWatch is enabled. Will be true by default when this
+         * configuration is included in project.
+         *
+         * @param exportConfigurationProperties The export configuration
+         * @param cloudWatchAsyncClient  The CloudWatch async client
+         * @return A CloudWatchMeterRegistry
+         */
     @Singleton
     CloudWatchMeterRegistry cloudWatchMeterRegistry(ExportConfigurationProperties exportConfigurationProperties,
                                                     CloudWatchAsyncClient cloudWatchAsyncClient) {

--- a/micrometer-registry-cloudwatch/src/test/groovy/io/micronaut/configuration/metrics/micrometer/cloudwatch/CloudwatchRegistryFactorySpec.groovy
+++ b/micrometer-registry-cloudwatch/src/test/groovy/io/micronaut/configuration/metrics/micrometer/cloudwatch/CloudwatchRegistryFactorySpec.groovy
@@ -46,9 +46,13 @@ class CloudwatchRegistryFactorySpec extends Specification {
         then:
         def list = context.getBeansOfType(CloudWatchAsyncClient)
         list.size() == 1
-        SdkClientConfiguration clientConfiguration = list[0].getProperties().get("clientConfiguration")
-        def properties = clientConfiguration.getProperties().get("attributes")["attributes"]
-        def result = properties.entrySet().stream().filter(x -> x.getValue().toString().startsWith("micronaut")).collect(Collectors.toList())
+
+        when:
+        SdkClientConfiguration clientConfiguration = list[0].properties.clientConfiguration
+        def properties = clientConfiguration.properties.attributes.attributes
+        def result = properties.findAll { it.value.value.toString().startsWith("micronaut") }
+
+        then:
         result.size() == 1
 
         cleanup:

